vowpalwabbit/cats.cc

```diff
@@ -61,6 +48,32 @@ int cats::learn(example& ec, experimental::api_status* status = nullptr)
   return error_code::success;
 }
 
+float cats::get_loss(const VW::cb_continuous::continuous_label& cb_cont_costs, float predicted_action) const
+{
+  float loss = 0.f;
+  if (!cb_cont_costs.costs.empty())
+  {
+    const float continuous_range = max_value - min_value;
+    const float unit_range = continuous_range / num_actions;
+
+    const float ac = (predicted_action - min_value) / unit_range;
+    int discretized_action = static_cast<int>(floor(ac));
+    // centre of predicted action
+    const float centre = min_value + discretized_action * unit_range + unit_range / 2.0f;
+
+    // is centre close to action from label
+    auto logged_action = cb_cont_costs.costs[0].action;
+    if ((logged_action - bandwidth <= centre) && (centre <= logged_action + bandwidth))
+    {
+      float actual_b = std::min(max_value, logged_action + bandwidth) - std::max(min_value, logged_action - bandwidth);
```

Comment:
I think we should replace logged_action with centre when computing actual_b. (I think this does make a small difference!)